viewing_party/party.py

```diff
@@ -1,23 +1,165 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    
+    movie = {}
+    movie_set = {title, genre, rating}
+
+    if None in movie_set:
+        return None
+    else:
+        movie["title"] = title
+        movie["genre"] = genre
+        movie["rating"] = rating
+    
+    return movie
+
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+
+    for movie in user_data["watchlist"]:
+        if title != movie["title"]:
+            continue
+        else:
+            movie_index = user_data["watchlist"].index(movie)
+            removed_movie = user_data["watchlist"].pop(movie_index)
+            add_to_watched(user_data, removed_movie)
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+
+    total = 0
+
+    if not user_data["watched"]:
+        return 0.0
+    
+    for movie in user_data["watched"]:
+        total += movie["rating"]
+    
+    average = total / len(user_data["watched"])

```

Comment:
👍 
Nice job doing the whole average calculation yourself! Now the next time you need an average, see if you can find some python library functions that might be helpful!

Reply:
**Notes for Myself**

Could use the sum function with list comprehension:
```
average = sum([movie["rating"] for movie in user_data["watched"]]) / len(user_data["watched"])
```